<launch>
	<arg name="world_file" default="trees.world" />

	<include file="$(find gazebo_ros)/launch/empty_world.launch">
		<arg name="world_name" value="$(find drone_swarm)/world/trees.world" />
		<arg name="paused" value="false" />
		<arg name="use_sim_time" value="true" />
		<arg name="gui" value="true" />
		<arg name="debug" value="false" />
	</include>
	
	<param name="robot_description" textfile="$(find drone_swarm)/urdf/drone.urdf" />
	
	<group ns="drone1">
		<param name="tf_prefix" value="drone1_tf" />
		
		<node name="spawn_drone1" pkg="gazebo_ros" type="spawn_model" args="-urdf -param /robot_description -x 10 -z 1 -model drone1 "> 
			<remap from="laser" to="drone1_laser"/>
		</node>
		
		<node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
		
		<node name="drone1_controller" pkg="drone_swarm" type="drone_controller.py" args="model:=drone1 target_pos:=10,10,15" output="screen"/>
	</group>
	
	<group ns="drone2">
		<param name="tf_prefix" value="drone2_tf" />
		
		<node name="spawn_drone2" pkg="gazebo_ros" type="spawn_model" args="-param /robot_description -urdf -z 1 -model drone2" />

		<node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
		
		<node name="drone2_controller" pkg="drone_swarm" type="drone_controller.py" args="model:=drone2" output="screen"/>
	</group>
	
	
	<node name="ground_control" pkg="drone_swarm" type="ground_control.py" args="amount_of_drones:=2"/>
	
	<node name="rviz" pkg="rviz" type="rviz"/>
</launch> 
